File: frontend/src/api/e2eTests.ts
import { request } from './core';

export type E2ETestRunnerKey = 'spec' | 'partner';

export interface E2ETestRunnerOption {
  key: E2ETestRunnerKey;
  label: string;
  description: string;
}

export interface E2ETestMetadataResponse {
  runners: E2ETestRunnerOption[];
}

export interface E2ETestRunRequest {
  runner: E2ETestRunnerKey;
  output_tail_lines: number;
}

export interface E2ETestRunResponse {
  runner: E2ETestRunnerKey;
  success: boolean;
  exit_code: number;
  started_at: string;
  finished_at: string;
  duration_seconds: number;
  report_path: string | null;
  output_tail: string;
  report_excerpt: string | null;
}

export const e2eTestsApi = {
  getE2ETestMetadata: () => request<E2ETestMetadataResponse>('/e2e-tests/metadata'),
  runE2ETest: (payload: E2ETestRunRequest) =>
    request<E2ETestRunResponse>('/e2e-tests/run', {
      method: 'POST',
      body: JSON.stringify(payload),
    }),
};

File: backend/app/schema/e2e_tests.py
from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel

E2ETestRunnerKey = Literal["spec", "partner"]


class E2ETestRunnerOption(BaseModel):
    key: E2ETestRunnerKey
    label: str
    description: str


class E2ETestMetadataResponse(BaseModel):
    runners: list[E2ETestRunnerOption]


class E2ETestRunRequest(BaseModel):
    runner: E2ETestRunnerKey
    output_tail_lines: int = 160


class E2ETestRunResponse(BaseModel):
    runner: E2ETestRunnerKey
    success: bool
    exit_code: int
    started_at: datetime
    finished_at: datetime
    duration_seconds: float
    report_path: str | None
    output_tail: str
    report_excerpt: str | None
